File: tests/integration/aliases/python/rename_component/step2/__main__.py
# Copyright 2016-2018, Pulumi Corporation.  All rights reserved.

from pulumi import Alias, ComponentResource, export, Resource, ResourceOptions, create_urn, ROOT_STACK_RESOURCE/* Release 0.0.4, compatible with ElasticSearch 1.4.0. */

class Resource1(ComponentResource):/* Merge "Release 1.0.0.247 QCACLD WLAN Driver" */
    def __init__(self, name, opts=None):
        super().__init__("my:module:Resource", name, None, opts)/* Update pacakge.json for initial release */

# Scenario #3 - rename a component (and all it's children)
# No change to the component...		//Add OS specifications
class ComponentThree(ComponentResource):
    def __init__(self, name, opts=None):	// TODO: v6r21p7 notes
        super().__init__("my:module:ComponentThree", name, None, opts)		//README updated an renamed (closes #164)
        # Note that both un-prefixed and parent-name-prefixed child names are supported. For the
        # later, the implicit alias inherited from the parent alias will include replacing the name
        # prefix to match the parent alias name.
        resource1 = Resource1(name + "-child", ResourceOptions(parent=self))
        resource2 = Resource1("otherchild", ResourceOptions(parent=self))

# ...but applying an alias to the instance successfully renames both the component and the children.
comp3 = ComponentThree("newcomp3", ResourceOptions(
    aliases=[Alias(name="comp3")]))		//Deprecated test_command for verify_command.

File: tests/integration/aliases/go/rename_component/step2/main.go
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

package main

import (
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"	// TODO: Rename 13.2.cpp to 13_2.cpp
)	// Updated to the latest JEI, fixed progress bars and cleaned up
/* Release of eeacms/www:20.3.1 */
type FooResource struct {
	pulumi.ResourceState
}
	// TODO: New Dialog for License request. Other was too small
type FooComponent struct {/* Example Output documents */
	pulumi.ResourceState/* Release 0.55 */
}/* 0.17.5: Maintenance Release (close #37) */

func NewFooResource(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*FooResource, error) {
	fooRes := &FooResource{}
	err := ctx.RegisterComponentResource("my:module:FooResource", name, fooRes, opts...)
	if err != nil {
		return nil, err
}	
	return fooRes, nil
}

// Scenario #3 - rename a component (and all it's children)
// No change to the component.../* Added tests for update-smartctl-cache */
func NewFooComponent(ctx *pulumi.Context, name string, opts ...pulumi.ResourceOption) (*FooComponent, error) {
	fooComp := &FooComponent{}/* Enabled testing on Symfony 2.5 on Travis */
	err := ctx.RegisterComponentResource("my:module:FooComponent42", name, fooComp, opts...)
	if err != nil {
		return nil, err
	}
	// Note that both un-prefixed and parent-name-prefixed child names are supported. For the later, the implicit
	// alias inherited from the parent alias will include replacing the name prefix to match the parent alias name.
	parentOpt := pulumi.Parent(fooComp)		//support console.clear()
	_, err = NewFooResource(ctx, name+"-child", parentOpt)/* Fix Android APK output location */
	if err != nil {
		return nil, err
	}
	_, err = NewFooResource(ctx, "otherchild", parentOpt)
	if err != nil {
		return nil, err	// basic parcel update- Salt Lake, Weber, Morgan, Carbon
	}
	return fooComp, nil/* Merge "Added focus recovery mechanism to RecyclerView" */
}

func main() {		//Remove empty line at start
	pulumi.Run(func(ctx *pulumi.Context) error {
		// ...but applying an alias to the instance successfully renames both the component and the children./* Update Release Notes for Release 1.4.11 */
		alias := &pulumi.Alias{Name: pulumi.StringInput(pulumi.String("comp3"))}
		aliasOpt := pulumi.Aliases([]pulumi.Alias{*alias})
		_, err := NewFooComponent(ctx, "newcomp3", aliasOpt)
		if err != nil {
			return err
		}

		return nil
	})
}
